### PR TITLE
Hog rewrite

### LINKED ISSUE #1828: HoG bugs v2.0
1)gods didn't spawn what the total fuck-->I'm a dumb fuck,fixed
2)weird god icon when transformed through TP-->possibly fixed, might need some more info
3)prophet didn't spawn with his hat
4)friendly thing to projectiles and not to turrets themself
5)godspeak prophet ability linked to the hat,fuckin shit,should be linked to the prophet himself
6)make it possible to convert enemy followers
7)hugely nerf claymore, it encourages killing which is not the point of the gamemode at all,or actually make the equipment more "defense"/"kidnap" based instead of lolkill
8)godspeak ability is working funnily,check that shit
9)if a god dies all its followers should become normal people again

### PR DESCRIPTION
**Pull request by Carbonhell**
Fixes #1828 
adds forge, lazarus and translocator
i had to edit so much shit,gimme time to fully list all the changes:

1)HUD is not defined because it is created manually for each team when one is generated
2)Added a proc, compareAllValues(list/A, list/B), checks if the values associated to the elements of the first list are all bigger or equal than the ones of the second(Only works with lists with number values!), used in forge
3)Improved traitor panel HOG options, you can make someone part of a team(Red and blue for now) as a follower, then the follower|prophet|god options show up,along with a Clear option to de-hogify someone.
4)var/global/list/teams = list("red", "blue"). Any amount of teams can be added by just adding a team(MUST be a color, such as "green"), and the code will handle the new team by itself. it could be possible to edit this at roundstart too but i'm not too sure,we'll test
5)The teams are managed in the mob's faction. Mobs now start with a faction element called "team", which then HoG gamemode associates to the color, such as "red". Then the proc is_in_any_team(mob/A) can be used to easily see what team A is in.
6)Rank(follower|prophet|god) is instead dealt with in the mind special role. the proc what_rank(datum/mind/A) deals with it and returns 0 if the mind is not an HoG player, 1 if it's a follower,2 if prophet and 3 if god.
7)get_gods() returns a list of all the living gods at the moment of the call,since dead ones get deleted from the game. get_team_god(team) returns only the god of said team.
8)get_team_followers(team) returns a list containing all the mobs(because thanks to the faction system even a goliath can be an HoG player) who are in that team.
9)Added the structures lacking the sprites such as Forge, which is used to craft hog weapons/clothings/banners similar to an autolathe; Lazarus statue which revives friends; Translocator which is used by the god to move his nexus(Translocator gets deleted after use.)
10)Added an hog themed scythe/dagger, which are like a fireaxe and a combat knife,slightly nerfed through. Also nerfed claymore to make players understand that the point of the game is converting, since killing doesn't reward you anything.
11)CLOTHINGS CURRENTLY LACK AN ONMOB SPRITE due to a missing tg proc which i have to port in a separate pr,but it's just an aesthetical bug.
12)Godspeak ability is now linked to the prophet and not by his hat, through prophet hat is very strong, stronger than the follower hat.
13)Defense pylons are smart now: they only target people not in their same team, and even if they happen to shoot a friendly structure/mob, the beam won't hurt them.
14)Href protection in forge to prevent fun memers to be able to make anything they wish

And last but not for importance, an huge thanks to Fury for all the sprites!Beautiful as always.
And an huge thanks to tg coders such as antur, neersighted, dunc and others for helping me with the various troubles in the way!

### CODE REVIEW

**Comment by FuryMcFlurry** [comment]
Awww yea, resprited most shit to allow more teams. it better be worth it

File: code/datums/mind.dm [hunk 1]
```diff
@@ -1253,43 +1252,37 @@
...
+			ticker.mode.remove_hog_follower(src, 0)
+			current << "<span class='danger'><B>You have been brainwashed... again! Your faith is no more!</B></span>"
+			message_admins("[key_name_admin(usr)] has de-hand of god'ed [current].")
+			log_admin("[key_name(usr)] has de-hand of god'ed [current].")
+
+		else if(href_list["handofgod"] in teams)
+			make_Handofgod_follower(href_list["handofgod"])
+			message_admins("[key_name_admin(usr)] has red follower'ed [current].")
+			log_admin("[key_name(usr)] has red follower'ed [current].")
+
```
Comment (Kokojos) [inline code/datums/mind.dm:1265]: Might get confused with ''Admin has followed user'' from when we orbit and stuff.

Comment (Carbonhell) [inline code/datums/mind.dm:1265]: What do you suggest then?Perhaps "x has added y to the 'side' hand of god team as a 'rank'?

Comment (Carbonhell) [inline code/datums/mind.dm:1265]: Fixed

**Comment by danhibiki** [comment]
Sure, but lets make sure we get those clothes fixed as soon as possible.

**Comment by Nokington** [comment]
eyha ures

**Comment by Carbonhell** [comment]
2 headmins' ok, now all i need is a codecheck and this can be merg'd

**Pull request merged**